### ISSUE TITLE
Can I use ff_api from forked process?

### ISSUE THREAD

**Issue by ghost** [F-Stack/f-stack#400]
If I understand correctly, the F-Stack environment can be initialized only in main thread, so it's impossible to use ff_api in the other threads, right? But how about forked processes?

**Comment by jfb8856606** [comment]
Yes, you are right. ff_api only can be called that run `ff_init()`‘s thread.

Forked processes, it no tested, i guess it can't be also.

**Comment by jfb8856606** [comment]
Update: F-Stack now supports forked processes. Several improvements have been made since the original reply:

- **2023-04**: Added `FF_MULTI_SC` macro to let child process workers inherit the specified sc (commit 3240dd0). Also added refcount in sc for fork and detach (commit ac0321e).
- **2025-05**: Full fork support added — each process now has its own FreeBSD `struct thread`, similar to the Linux kernel model (commit 4891fab).

Additionally, F-Stack's **LD_PRELOAD** (`ff_hook_syscall`) functionality is now supported, which allows existing applications to use F-Stack's network stack transparently without code modification — including forked child processes.

So the original answer of "it can't work with fork" is now outdated. Both direct `ff_api` usage in forked processes and the LD_PRELOAD approach are functional in current versions.

Closing as the original question has been answered and the feature has been implemented.
